### PR TITLE
docs: partial reports can be submitted

### PR DESCRIPTION
**Pull request by mlavacca**
**What type of PR is this?**

/kind documentation
/area conformance

**What this PR does / why we need it**:

After discussing the possibility of accepting partial reports in https://github.com/kubernetes-sigs/gateway-api/pull/3021, we eventually decided to encourage implementations to start submitting reports even if there is no full conformance. This PR updates the documentation to reflect this final decision.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by shaneutt — APPROVED** [review]
After thinking it over, I do agree that it's better to get partial reports rather than no reports because this will increase our awareness of implementations in the field and enable more opportunities to engage with them.

I do think we may need to adjust the display to accommodate (make sure it's clear, and perhaps delineated when an implementation is reporting partial). I approve,

/approve

However I would like us to make sure we have a follow-up issue in place for considering how the display should look like for this (I don't consider it a blocker though) before we merge:

/hold

**Comment by mlavacca** [comment]
> After thinking it over, I do agree that it's better to get partial reports rather than no reports because this will increase our awareness of implementations in the field and enable more opportunities to engage with them.
> 
> I do think we may need to adjust the display to accommodate (make sure it's clear, and perhaps delineated when an implementation is reporting partial). I approve,
> 
> /approve
> 
> However I would like us to make sure we have a follow-up issue in place for considering how the display should look like for this (I don't consider it a blocker though) before we merge:
> 
> /hold

Yes, it makes sense. I've opened https://github.com/kubernetes-sigs/gateway-api/issues/3183 to capture this. Let me know if it fits with what you had in mind :)

**Comment by shaneutt** [comment]
LGTM

/unhold
/cc @robscott @youngnick

File: conformance/reports/README.md
Comment (robscott) [inline conformance/reports/README.md:124]: It seems like we probably need to include some guidelines for how we handle `partial` conformance results. Maybe it's sufficient to just say that they'll be displayed separately?

File: conformance/reports/README.md
Comment (robscott) [inline conformance/reports/README.md:127]: I think there's another reason for partial conformance - the tests may all pass, but the reproduction required some kind of additional unexpected steps.

**Review by robscott — COMMENTED** [review]
Thanks @mlavacca! LGTM, just a couple nits where further clarification could be useful.

File: conformance/reports/README.md
Comment (mlavacca) [reply conformance/reports/README.md]: 👍  I reworded this section to include this information

**Comment by youngnick** [comment]
All of @robscott's feedback is addressed, going to get this one in.

/lgtm

**Pull request merged**
